lib/phoenix/socket.ex

```diff
@@ -589,6 +590,9 @@ defmodule Phoenix.Socket do
             :error
         end
 
+      {:error, _reason} = err ->
+        err
+
       :error ->
         :error
```

Comment:
Let's rewrite this to `{:error, :undefined}`.

Reply:
I'm a little worried about this change since it seems to break backward-compatibility? I don't know if people rely on this returning just `:error`.

Reply:
To be clear, we will do this:

```
:error ->
  {:error, :undefined}
```

So internally, the `error_handler` always has a reason. The socket users themselves can still return `:error`.

Reply:
Actually, no. You are correct, this would be a backwards incompatible change for people implementing custom transports. Gah, I am so sorry.

I think we will have to revert the `:error -> {:error, :undefined}` changes.

Here is my (hopefully) last proposal:

1. Allow `{:error, reason}`
2. The custom error handler will only be called if `{:error, reason}` is returned

Reply:
Only saw this now. No worries! Made the revert, took me 5 minutes :)